components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -74,6 +78,38 @@ brave_wallet::SwapController* BraveWalletService::swap_controller() const {
   return swap_controller_.get();
 }
 
+std::vector<std::string> BraveWalletService::WalletAccountNames() const {
+  std::vector<std::string> account_names;
+  for (const auto& account_name_value :
+       prefs_->Get(kBraveWalletAccountNames)->GetList()) {
+    const std::string* account_name = account_name_value.GetIfString();
+    DCHECK(account_name) << "account name type should be string";
+    account_names.push_back(*account_name);
+  }
+  return account_names;
+}
+
+// TODO(Douglas): This need to be seperated into 2 separate methods
```

Comment:
Please also create a follow-up issue for it so we don't forget.

Reply:
Issue Created https://github.com/brave/brave-browser/issues/16887